### PR TITLE
Prepare `bevy_smud` for 0.11 `bevy`

### PR DESCRIPTION
**Pull request by StaffEngineer**
This is just partial update of repo to work with bevy's 0.11 version. This PR can be used as reference/starting point for updating the project to 0.11.

### CODE REVIEW

File: assets/shapes.wgsl [hunk 1]
```diff
@@ -44,7 +45,7 @@ fn ndot(a: vec2<f32>, b: vec2<f32>) -> f32 {
     return a.x * b.x - a.y * b.y;
 }
 
-fn dot2(a: vec2<f32>) -> f32 {
+fn dot2_(a: vec2<f32>) -> f32 {
```
Comment (johanhelsing) [inline assets/shapes.wgsl:48]: Is there now s built in dot2? Could we use it instead?

File: assets/shapes.wgsl [hunk 2]
```diff
@@ -565,7 +566,7 @@ fn sd_exponential_falloff(d: f32, size: f32, power: f32) -> f32 {
     return a;
 }
 
-fn sd_exponential_falloff_3(d: f32, size: f32) -> f32 {
+fn sd_exponential_falloff_3_(d: f32, size: f32) -> f32 {
```
Comment (johanhelsing) [inline assets/shapes.wgsl:569]: Why the trailing underscores?

File: src/sdf_assets.rs [hunk 3]
```diff
@@ -15,26 +15,39 @@ pub trait SdfAssets {
...
+                r#"
+#define_import_path bevy_smud::fill{id}
+
 fn fill(d: f32, color: vec4<f32>) -> vec4<f32> {{
     {body}
 }}
 "#
-        ));
+            ),
+            file!(),
```
Comment (johanhelsing) [inline src/sdf_assets.rs:49]: What's this about?

File: assets/shapes.wgsl (hunk 1 above)
Comment (StaffEngineer) [reply assets/shapes.wgsl:48]: Could you paraphrase the question?

Shader functions can't end on number anymore due to https://github.com/bevyengine/naga_oil usage.

File: assets/shapes.wgsl (hunk 2 above)
Comment (StaffEngineer) [reply assets/shapes.wgsl:569]: Shader functions can't end on number anymore due to https://github.com/bevyengine/naga_oil usage.

File: src/sdf_assets.rs (hunk 3 above)
Comment (StaffEngineer) [reply src/sdf_assets.rs:49]: AFAIK this field for debugging purposes, to see "filename" in logs. file! will name it `sdf_assets.rs` but I think it's better just put `bevy_smud::fill` instead.

File: src/lib.rs [hunk 4]
```diff
@@ -341,16 +346,29 @@ fn extract_sdf_shaders(mut main_world: ResMut<MainWorld>, mut pipeline: ResMut<S
...
+
+@fragment
+fn fragment(in: FragmentInput) -> @location(0) vec4<f32> {{
+    let d = sdf::sdf(in.pos);
+    return fill::fill(d, in.color);
+}}
 "#
-            ));
+                ),
+                file!(),
```
Comment (johanhelsing) [inline src/lib.rs:370]: should perhaps include the generated id here as well

File: src/sdf_assets.rs [hunk 5]
```diff
@@ -15,26 +15,39 @@ pub trait SdfAssets {
...
+
+#import bevy_smud::shapes as shapes
+
 fn sdf(p: vec2<f32>) -> f32 {{
     {body}
 }}
 "#
-        ));
+            ),
+            file!(),
```
Comment (johanhelsing) [inline src/sdf_assets.rs:31]: ditto

**Pull request closed without merging**